src/views/components/model.jsx

```diff
@@ -1,17 +1,64 @@
 import React from 'react';
-
+import ModelStore  from '../../store/model-store.js';
 export default class Model extends React.Component {
 
-    render() {
-        if (this.props.model === null) {
-            return <i>No model found</i>;
+    constructor(props){
+        super(props);
+        if(props.model !== undefined)
```

Comment:
Original way of me checking which didn't get changed. I'll change this to what you described earlier but for models obviously. FIXED.